packages/tab/package.json

```diff
@@ -10,15 +10,16 @@
   "module": "dist/esm/index.js",
   "scripts": {
     "build": "run-s build:js build:cjs build:css",
-    "build:cjs": "babel --root-mode upward src --out-dir dist/cjs",
+    "build:cjs": "babel --root-mode upward src --out-dir dist/cjs && cp src/react/index.d.ts dist/cjs/react/index.d.ts && cp src/index.d.ts dist/cjs/index.d.ts",
     "build:css": "build-css --useGlamor -i dist/cjs/css/index.js",
-    "build:js": "cross-env ESM=true babel --root-mode upward src --out-dir dist/esm",
+    "build:js": "cross-env ESM=true babel --root-mode upward src --out-dir dist/esm && cp src/react/index.d.ts dist/esm/react/index.d.ts && cp src/index.d.ts dist/esm/index.d.ts",
```

Comment:
Is it important to have typings in `esm` alongside the `cjs` dir? 

I would assume the `"types"` property in the package.json (below) controls where tooling looks for typings.

Reply:
I'm not sure how to test with `esm`, or if it even makes sense to use them with TS. (What's the main benefit? Tree-shaking?)

We should probably just take them out of the `esm` folder until we know of a concrete use case for them there.

Reply:
Removed

Reply:
Hey Justin, just getting back in...

Yes, we have ESM as a format for tree shaking, mostly.  In the future, in-browser module loading is also a potential.

So, I suppose I'm a little confused and also probably didn't clearly articulate my question.  I'll try again.  Let me know how I do:

If the .d.ts file goes to npm *somewhere*, I would think that *where* would be pointed to by the `typings` package.json property -- and that alone would be sufficient for tooling/TS to pickup.  This is regardless of whether one is consuming the cjs or esm format of the DS code.  

Do you think that's ^ the case?

To your other question, yes, I *think* projects that use the ESM format would likely enjoy the TS typings as well.  TS looks more like ESM than CJS -- I would have almost thought that TS would use the ESM format, pointed to by the `module` property in package.json.

Reply:
Got it, that makes sense. I was able to verify that your suggestion works for `cjs`. I'm not sure how to test with `esm`. Pushed up another commit with this.

Reply:
For me at least, TypeScript is resolving `import Tab from '@pluralsight/ps-design-system-tab'` to `/dist/cjs`, so it seems to be following `main`.

Reply:
The only way I can think of to get the `esm` is to do this:

`import Tab from '@pluralsight/ps-design-system-tab/dist/esm'`

If I do this, then it doesn't find typings if they're coming from `src`. If I place the `.d.ts` files alongside the source in `dist/esm`, then it works.

Let me know if you have a better way to test this.

Reply:
Hey Justin, again, sorry to be slow.  I've been in and out, mostly out. Thanks for helping me through on this.  You're a trooper, as always.  I miss working with you more.

I think you're testing is great.  It's good to know what format TS is resolving, using what package.json fields.  I don't have any further tests or testing methods I have thought of.

Reply:
@jaketrent 

All right, I have a solution that seems to check all my boxes:

Easy for consumers to add props via interface augmentation:

```typescript
declare module '@pluralsight/ps-design-system-tab' {
    interface ListItemProps {
        name: string                          //add "name" prop to ListItem
    }
}
```

Typings work with both `cjs` and `esm`

Only need to copy out a single file to the `dist` for each, like so:

```shell
cp src/index.d.ts dist/cjs/index.d.ts`
```
and
```shell
cp src/index.d.ts dist/esm/index.d.ts`
```

This is because I put the declaration straight into the "public interface" `.index.d.ts`